### PR TITLE
Jjp241 Data_Processor_Block

### PR DESCRIPTION
**Pull request by jakub-podolak**
Useful piece of code

### CODE REVIEW

File: code/data_processor_block/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Data Processor Block
+#### Takes packets with data from base_parser and sends it to graphyte
+### Run it:
+`python data_processor.py config.ini`
+### Config File:
+`config.ini` - You specify here your graphite address (if you host it on the same device its `localhost`), graphyte prefix (kinda like folder with your data in graphyte) and sleep_time
```
Comment (evemorgen) [inline code/data_processor_block/README.md]: It's `Graphite` not `Graphyte` 🤣

File: code/data_processor_block/data_processor.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+def run():
+    directories = get_directories(config)
+
+    while True:
+        try:
+            for input_log in directories:
+                file = open(input_log, "r")
+                file.seek(0,2)
+                file.seek(file.tell()-41, 0) #Packet should have a size of 40 characters
```
Comment (evemorgen) [inline code/data_processor_block/data_processor.py]: how do we know whether there were just one line added to the file? This approach might result in skipping lines, which we should avoid I believe?

File: code/data_processor_block/data_processor.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+graphyte.init(graphyte_address, prefix=graphyte_prefix)
+
+#open log file
+logging.basicConfig(filename="data_processor.log",level=logging.INFO)
+
+#function to read paths to input logs from config file
+def get_directories(config):
+    paths = config.items("PATHS")
+    directories = [second for first,second in paths]
```
Comment (evemorgen) [inline code/data_processor_block/data_processor.py]: this is just cosmetics, but we don't need to assign this list to variable if we're not gonna use that anywhere else, that means we could just return the list comprehension like this:
```python
def get_directories(config):
	    paths = config.items("PATHS")
	    return [second for first, second in paths]
```

File: code/data_processor_block/data_processor.py (hunk 2 above)
Comment (jjp241) [reply code/data_processor_block/data_processor.py]: TBH I'm afraid, that any kind of going back in the log file is potentially dangerous. Imagine a situation when our base station receives packets more frequently than we can send it to graphite. In such a scenario, we are not displaying current values, but can actually get stuck with old data. In my opinion printing current data is more important than sending everything.

File: code/data_processor_block/data_processor.py (hunk 3 above)
Comment (jjp241) [reply code/data_processor_block/data_processor.py]: Yes, of course. I have some kind of instinct to make everything legible, sometimes in the wrong way making things actually too complicated.

File: code/data_processor_block/README.md (hunk 1 above)
Comment (jjp241) [reply code/data_processor_block/README.md]: Aye DJ Evening-Morgen, but how do you exblain that python lib is called `graphyte` ? Hmm? Cocoa Clogged?

File: code/data_processor_block/data_processor.py (hunk 2 above)
Comment (evemorgen) [reply code/data_processor_block/data_processor.py]: If we store hard copy of received packets, I totally agree

File: code/data_processor_block/README.md (hunk 1 above)
Comment (evemorgen) [reply code/data_processor_block/README.md]: Here we go again,
`Graphite` - metric store engine
`Graphyte` - Python module name
So when you `prefix` metrics, you prefix them in datastore or library itself? I bet it is the first one. Probably for the simplicity sake guys from `graphyte` called it just [`prefix`](https://github.com/Jetsetter/graphyte/blob/master/graphyte.py#L30)
And when it comes to defining tree (folder) structure you definetly do that in `Graphite` :P
So something is no yes here, cocoa not clogged at all!

File: code/data_processor_block/data_processor.py (hunk 2 above)
Comment (SlupikB) [reply code/data_processor_block/data_processor.py]: yup, all received packets are hard logging'ed

File: code/data_processor_block/README.md (hunk 1 above)
Comment (jjp241) [reply code/data_processor_block/README.md]: Okay, cosmetic fix, check out 0fbbb63

File: code/data_processor_block/data_processor.py (hunk 3 above)
Comment (jjp241) [reply code/data_processor_block/data_processor.py]: Resolved here: ed4701c

**Comment by jjp241** [comment]
New approach to create packets - described in Readme of data_processor and merge_block

**Review by evemorgen — APPROVED** [review]

**Pull request merged**